packages/vuetify/src/composables/filter.ts

```diff
@@ -0,0 +1,86 @@
+
+// Utilities
+import { getPropertyFromItem, propsFactory, wrapInArray } from '@/util'
+import { computed } from 'vue'
+
+// Types
+import type { PropType, Ref } from 'vue'
+
+export type FilterFunction = typeof defaultFilter
+export type FilterMode = 'intersection' | 'union'
+
+export interface FilterProps {
+  filterFn?: FilterFunction
+  filterMode?: FilterMode
+}
+
+// Composables
+export function defaultFilter (text: string, query?: string) {
+  if (typeof query !== 'string') return true
+  if (typeof text !== 'string') return false
+
+  return text.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+}
+
+export const makeFilterProps = propsFactory({
+  filterFn: Function as PropType<FilterFunction>,
+  filterMode: {
+    type: String as PropType<FilterMode>,
+    default: 'intersection',
+  },
+}, 'filter')
+
+export function filterItems (
+  items: Record<string, any>[],
+  keys: string[],
+  query?: string,
+  filter?: {
+    default?: FilterFunction
+    mode?: FilterMode
+    keyFilters?: Record<string, FilterFunction>
```

Comment:
Thinking of how we could support both complete custom filter fn and custom default key filter fn with a decent API, I wonder if we could do something like

```
// props
filterFn: {
  type: Function,
  default: defaultFilterFn
},
keyFilters: {
  type: Object as Prop<Record<string | Symbol, Function>,
}
```

```
const defaultKeyFilters = {
  [Symbol.for('some-key')]: defaultKeyFilterFn
}
```

this gets merged with the user provided one from `keyFilters`. so that users can both add their own key filter fns and replace the default one through a single prop.

Reply:
Do we want to make keyFilters part of public props?

Reply:
I just used keyFilters cause that's what you're calling it atm. I think it would make sense to call the prop the same across components if it does the same thing.

Reply:
Could you take a stab at implementing the functionality that you're describing?